File: experiments/llm-reasoning-cost/generated_code/python/05_file_io_codex.py
from pathlib import Path
from typing import Union


def write_greeting(filename: Union[str, Path]) -> tuple[bool, Exception | None]:
    try:
        with open(filename, "w", encoding="utf-8") as f:
            f.write("Hello from the program!")
        return True, None
    except Exception as e:
        return False, e

File: experiments/llm-reasoning-cost/generated_code/rust/05_file_io_codex.rs
use std::fs::File;
use std::io::{self, Write};
use std::path::Path;

pub fn write_greeting<P: AsRef<Path>>(filename: P) -> io::Result<()> {
    let mut file = File::create(filename)?;
    file.write_all(b"Hello from the program!")?;
    file.flush()?;
    Ok(())
}